api/app/controllers/spree/api/v2/platform/products_controller.rb

```diff
@@ -18,6 +18,14 @@ def scope_includes
           def allowed_sort_attributes
             super << :available_on
           end
+
+          def sorted_collection
+            collection_sorter.new(collection, current_currency, params, allowed_sort_attributes).call
+          end
+
+          def collection_sorter
+            Spree::Api::Dependencies.storefront_products_sorter.constantize
```

Comment:
@szymoniwacz let's add `platform_products_sorter` in https://github.com/spree/spree/blob/main/api/app/models/spree/api_dependencies.rb